tests/unit/utils/txn_parser/test_get_order_book_changes.py

```diff
@@ -0,0 +1,162 @@
+                ],
+            },
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_offer_partially_filled_and_filled(self: TestGetOrderBookChanges):
+        actual = get_order_book_changes(offer_partially_filled_and_filled["meta"])
+        expected = [
+            {
+                "account": "rNzgS71DyJPMnWMA8aS7NqvXP7bNuwyaZo",
+                "offer_changes": [
+                    {
+                        "direction": "sell",
+                        "quantity": {
+                            "currency": "USD",
+                            "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
+                            "value": "-63.7479881398749",
+                        },
+                        "total_price": {
+                            "currency": "USD",
+                            "issuer": "rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+                            "value": "-62.4730283770749",
+                        },
+                        "sequence": 5931,
+                        "status": "filled",
+                        "maker_exchange_rate": "0.98",
+                    }
+                ],
+            },
+            {
+                "account": "rPu2feBaViWGmWJhvaF5yLocTVD8FUxd2A",
+                "offer_changes": [
+                    {
+                        "direction": "sell",
+                        "quantity": {
+                            "currency": "USD",
+                            "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
+                            "value": "-117.3895136925395",
+                        },
+                        "total_price": {
+                            "currency": "USD",
+                            "issuer": "rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B",
+                            "value": "-115.0877585220975",
+                        },
+                        "sequence": 67701941,
+                        "status": "partially-filled",
+                        "maker_exchange_rate": "0.9803921568627452",
+                    }
+                ],
+            },
+        ]
+        self.assertEqual(actual, expected)
+
+    def test_offer_cancelled(self: TestGetOrderBookChanges):
+        actual = get_order_book_changes(offer_cancelled["meta"])
+        expected = [
+            {
+                "account": "rEUt5Wy44vDKBDaGkUWG6oSTvxmqgnKWCg",
+                "offer_changes": [
+                    {
+                        "direction": "buy",
+                        "quantity": {
+                            "currency": "XRP",
+                            "value": "47.504858",
+                        },
+                        "total_price": {
+                            "currency": "XDX",
+                            "issuer": "rMJAXYsbNzhwp7FfYnAsYP5ty3R9XnurPo",
+                            "value": "82335.52909",
+                        },
+                        "sequence": 70922543,
+                        "status": "cancelled",
```

Comment:
Just having a single status that's mandatory to understand the results is a bit risky since it's easy to overlook that field. (Which in this case would cause a user to interpret this as a new "Buy" offer for the amounts specified above, instead of treating this as a removal of a previous "Buy" offer)

Similar to the above comment, is there a way to convey the actual effect of a cancellation using the `quantity` and `total_price` fields?

Reply:
> …. You're right the amount should always be negative if the `status` is not "created".

This should be solved with my comment above. Then you would always have the actual amount that got changed.